### DISCUSSION TITLE
v3.0.0-beta1

### DISCUSSION THREAD

**Discussion by danharrin** [filamentphp/filament#7060] [General]
If you're interested in testing v3 beta, thank you so much for taking part! You will be helping us by ensuring that v3 is stable enough for public use and release on the 1st of August.

You can find the beta docs at https://beta.filamentphp.com. The [upgrade guide](https://beta.filamentphp.com/docs/3.x/panels/upgrade-guide) for the admin panel from v2, can be found on there. To install it, you'll need to use `"minimum-stability": "dev"` in your `composer.json` file - newer installations of Laravel use `stable`.

> It's come to our attention that Composer is installing alpha versions of v3 instead of beta, leading to weird experiences. You can check with `php artisan about`. If that's the case, in `composer.json` use `^3.0@beta` as the version constraint and it should be fixed.

However, it's best that we set some of your expectations, to ensure that you have a good experience, especially when interacting with project maintainers.

1) This is a very busy period for core maintainers of Filament. There are 5 people on the team, and we have full time jobs alongside Filament. Please be nice.
2) Things might take a little longer to process than you want or expect.
3) Support, especially on GitHub and Discord, will be much slower, or non-existent.
4) The experience using the Filament v3 beta may be substandard, buggy, or missing features.
5) The documentation may be wrong, or missing vital information.
6) The upgrade experience from v2 may be slow and painful. Especially if you have heavy customization.
7) If you're not reporting things in GitHub issues, they will likely get lost. Especially if you attempt to report bugs on Discord or in direct messages with maintainers. Those bugs are not tracked and will not be fixed.
8) This applies to all GitHub issues, but if you make no effort to provide a reproduction repository for an issue, it will be closed and not investigated. Please reproduce issues in a new project, and not your existing project. We are not interested in code that is not directly required to replicate the problem.
9) It's very likely that third party plugins will not be compatible with v3 yet. Be nice to plugin authors as well!
10) We expect you to submit pull requests for typos, especially to the documentation. Please do not open issues for these, the core team needs to focus on issues with the actual codebase.

If any of these things disappoint you, please do not use Filament v3 beta!

Now that we've set some expectations, I'd like to outline the priorities of Filament team members during the beta period:

1) Supporting v2 plugin authors to upgrade to v3. If you are a v2 plugin author and require support, please contact the core team in the `#plugin-developers-chat` channel on our Discord server. You will need to resubmit your plugins to our website as there are new quality guidelines to follow - [the process is documented here](https://github.com/filamentphp/filamentphp.com/tree/new-design#contributing). All PRs should be made to the `new-design` branch on the `filamentphp.com` repo.
2) Fixing critical bugs, especially those that did not exist in the same functionality of v2. Bugs in new v3 features will likely not be a priority to fix.
3) Improving the upgrade experience by adding rules to the automated upgrade script.

We are not prioritising adding new features to v3 at this time. If you submit a PR or make a feature request, it's quite likely that it will not be merged for weeks, especially if it is large.

Again, thank you for helping us out and respecting our time. We hope that you enjoy trying out all the new stuff we have in store, and love the new design!

<hr /><em>This discussion was created from the release <a href='https://github.com/filamentphp/filament/releases/tag/v3.0.0-beta1'>v3.0.0-beta1</a>.</em>

**Comment by mstfkhazaal** [comment]
when i run composer require filament/filament:"^3.0"
i get this error

Nothing to modify in lock file
Installing dependencies from lock file (including require-dev)
Nothing to install, update or remove
Package league/uri-parser is abandoned, you should avoid using it. No replacement was suggested.
Package tgalopin/html-sanitizer is abandoned, you should avoid using it. Use symfony/html-sanitizer instead.
Generating optimized autoload files
> Illuminate\Foundation\ComposerScripts::postAutoloadDump
Script Illuminate\Foundation\ComposerScripts::postAutoloadDump handling the post-autoload-dump event terminated with an exception

```
PHP Fatal error:  Declaration of Symfony\Component\String\UnicodeString::replaceMatches(string $fromRegexp, callable|string $to): static must be compatible with Symfony\Component\String\AbstractUnicodeString::replaceMatches(string $fromRegexp, $to): Symfony\Component\String\AbstractString in /Users/rdms/Desktop/Web/filament/v3/nscourse/vendor/symfony/string/UnicodeString.php on line 248

Fatal error: Declaration of Symfony\Component\String\UnicodeString::replaceMatches(string $fromRegexp, callable|string $to): static must be compatible with Symfony\Component\String\AbstractUnicodeString::replaceMatches(string $fromRegexp, $to): Symfony\Component\String\AbstractString in /Users/rdms/Desktop/Web/filament/v3/nscourse/vendor/symfony/string/UnicodeString.php on line 248

```

**Reply by miomech** [reply]
@mstfkhazaal 

Try:

1. Delete `/vendor` and `composer.lock` file.
2. Edit your `composer.json` and remove all filament 2 packages along with livewire v2 package as well.
3. In your `composer.json` add the package `"filament/filament": "^v3.0.0-beta1"
4. Save your changes
5. Run `composer install` 

Hopefully that helps.

**Reply by mstfkhazaal** [reply]
This is work right now 
i don't know if all thing is run perfectly

When login, dashboard not run 
i need to refresh the page to get dashboard

https://github.com/filamentphp/filament/assets/47450090/74850155-d225-4ade-840a-baa6bef52a7d

**Reply by miomech** [reply]
Hm, might be a bug? 

I' would check your dev console tools and see what the issue is. If it's an issue with filament you can submit it as a bug and help out the filament team fix it : ).

**Reply by mstfkhazaal** [reply]
Okay thanks I'll start experimenting and document all the bugs

**Reply by coolsam726** [reply]
I was also looking at this, it definitely has something to do with livewire's `$wire.navigate` or x-persist. I will keep digging into dev tools about this.
![image](https://github.com/filamentphp/filament/assets/5610289/149dcafc-14a3-42c1-9c2e-fae64200a1f7)

**Reply by coolsam726** [reply]
It may be related to [THIS](https://github.com/livewire/livewire/issues/5816) issue for livewire.

**Comment by aurawindsurfing** [comment]
Is there already availability for multiple panels? I assume yes?

Is there front facing panel option as well or was it not added yet?

Asking for a friend 😂

**Reply by stefanzweifel** [reply]
@aurawindsurfing You can run the following artisan command to create a new "app" panel.

```shell
php artisan make:filament-panel app
```

You can than configure a lot of stuff in the newly created `AppPanelProvider`. 
You can pass the name of the panel to the artisan command as well.

```shell
php artisan make:filament-resource User --panel=app
```

**Reply by aurawindsurfing** [reply]
Thank you. I could not find it in the docs anywhere

**Reply by aurawindsurfing** [reply]
@stefanzweifel where is it in the docs, please?

**Reply by mstfkhazaal** [reply]
https://beta.filamentphp.com/docs/3.x/panels/configuration

**Comment by killjin** [comment]
After using shortcut keys for global search and closing the global search results, the bound shortcut keys have no effect.

**Comment by datlechin** [comment]
I got JS error when trying to navigate to a page that has date time picker
look like the `x-load` doesn't works with `wire:navigate` because `date-time-picker.js` doesn't load when navigating without reloading the page

see the error log in below log file.

[127.0.0.1-1690112687714.log](https://github.com/filamentphp/filament/files/12138879/127.0.0.1-1690112687714.log)

**Reply by datlechin** [reply]
https://github.com/filamentphp/filament/assets/56961917/3bc05dd4-d99c-4f6c-8b50-b2766c8e87db

**Comment by datlechin** [comment]
on windows
<img width="268" alt="image" src="https://github.com/filamentphp/filament/assets/56961917/0a10a7ec-2c0d-4b7c-9c0c-462923443d92">

**Comment by datlechin** [comment]
Link in richeditor
<img width="647" alt="image" src="https://github.com/filamentphp/filament/assets/56961917/fa21cb76-48fc-418f-8167-c81ce310c6dc">

**Comment by danharrin** [comment]
@datlechin read the beta release notes in full. Issues and PRs are the only ways for us to track stuff.

**Comment by ramiropiro** [comment]
I'm gettin this right after login, then I refresh or press any buttom or menu and works with no error
![image](https://github.com/filamentphp/filament/assets/111913304/a4d2f0bc-8eeb-49e9-a1db-5f81dd4ab19d)

**Reply by danharrin** [reply]
Upgrade

**Comment by ali635** [comment]
when run this command "php artisan filament:install --panels" 
this error show me "  The "--panels" option does not exist.  "

**Reply by sinnbeck** [reply]
Version 3 is no longer in beta. So install the latest and if it does not work, please make a new discussion with your problem :)

**Comment by ali635** [comment]
when run this command "php artisan filament:install --panels" 
this error show me "  The "--panels" option does not exist.  "
